modules/login/login_module.e

```diff
@@ -0,0 +1,909 @@
+			package := "login"
+
+			create root_dir.make_current
+			cache_duration := 0
+		end
+
+feature {CMS_API} -- Module Initialization			
+
+	initialize (a_api: CMS_API)
+			-- <Precursor>
+		local
+			l_user_auth_api: like user_oauth_api
+			l_user_auth_storage: CMS_USER_OAUTH_STORAGE_I
+		do
+			Precursor (a_api)
+
+				-- Storage initialization
+			if attached {CMS_STORAGE_SQL_I} a_api.storage as l_storage_sql then
+				create {CMS_USER_OAUTH_STORAGE_SQL} l_user_auth_storage.make (l_storage_sql)
+			else
+				-- FIXME: in case of NULL storage, should Current be disabled?
+				create {CMS_USER_OAUTH_STORAGE_NULL} l_user_auth_storage
+			end
+
+				-- Node API initialization
+			create l_user_auth_api.make_with_storage (a_api, l_user_auth_storage)
+			user_oauth_api := l_user_auth_api
+		ensure then
+			user_oauth_api_set: user_oauth_api /= Void
+		end
+
+feature {CMS_API} -- Module management
+
+	is_installed (api: CMS_API): BOOLEAN
+			-- Is Current module installed?
+		do
+			Result := attached api.storage.custom_value ("is_initialized", "module-" + name) as v and then v.is_case_insensitive_equal_general ("yes")
+		end
+
+	install (api: CMS_API)
+		local
+			sql: STRING
+			l_setup: CMS_SETUP
+		do
+			l_setup := api.setup
+
+				-- Schema
+			if attached {CMS_STORAGE_SQL_I} api.storage as l_sql_storage then
+				if not l_sql_storage.sql_table_exists ("oauth2_gmail") then
+					--| Schema
+					l_sql_storage.sql_execute_file_script (l_setup.environment.path.extended ("scripts").extended ("core.sql"))
+
+					if l_sql_storage.has_error then
+						api.logger.put_error ("Could not initialize database for blog module", generating_type)
+					end
+				end
+				api.storage.set_custom_value ("is_initialized", "module-" + name, "yes")
+			end
+		end
+
+feature {CMS_API} -- Access: API
+
+	user_oauth_api: detachable CMS_USER_OAUTH_API
+			-- <Precursor>		
+
+
+feature -- Filters
+
+	filters (a_api: CMS_API): detachable LIST [WSF_FILTER]
+			-- Possibly list of Filter's module.
+		do
+			create {ARRAYED_LIST [WSF_FILTER]} Result.make (1)
+			if attached user_oauth_api as l_user_oauth_api then
+				Result.extend (create {OAUTH_GMAIL_FILTER}.make (a_api, l_user_oauth_api))
+			end
+		end
+
+feature -- Access: docs
+
+	root_dir: PATH
+
+	cache_duration: INTEGER
+			-- Caching duration
+			--|  0: disable
+			--| -1: cache always valie
+			--| nb: cache expires after nb seconds.
+
+	cache_disabled: BOOLEAN
+		do
+			Result := cache_duration = 0
+		end
+
+feature -- Router
+
+
+	setup_router (a_router: WSF_ROUTER; a_api: CMS_API)
+			-- <Precursor>
+		do
+			if attached user_oauth_api as l_user_oauth_api then
+				configure_web (a_api, l_user_oauth_api, a_router)
+			end
+		end
+
+
+	configure_web (a_api: CMS_API; a_user_oauth_api: CMS_USER_OAUTH_API; a_router: WSF_ROUTER)
+		do
+			a_router.handle_with_request_methods ("/roc-login", create {WSF_URI_AGENT_HANDLER}.make (agent handle_login (a_api, ?, ?)), a_router.methods_head_get)
```

Comment:
Could we groups all those route into a common sub path
such as

/account/roc-login
/account/roc-register
and so on ...

this way, we could use a starts_with mapping, and then use a new router, to dispatch them.
so instead of having a list of mappings, we would have a tree of mapping.

Maybe this is not a good idea, but maybe this would be a way later to improve performance.
And we could even have kind of delayed router setup.
like a vision2 tree, we can expand the subtree only when needed. Here we could do the same maybe.
BUT not for now, this is just an idea I wanted to share aloud.

For now, I would just found nicer to group under a /account/...
so that the api looks structured.